app/uk/gov/hmrc/individualincomedesstub/domain/Employment.scala

```diff
@@ -56,13 +57,18 @@ case class Employment(employerPayeReference: EmpRef,
                       endDate: Option[String],
                       payments: Seq[HmrcPayment],
                       payFrequency: Option[String] = None) {
-  def containsPaymentWithin(interval: Interval) =
-    payments.exists(_.isPaidWithin(interval))
+
+  private def containsPaymentWithin(interval: Interval) = payments.exists(_.isPaidWithin(interval))
+
+  def isWithin(interval: Interval) = (startDate, endDate) match {
+      case (Some(start), Some(end)) => interval.overlaps(toInterval(start, end)) || containsPaymentWithin(interval)
+      case _ => containsPaymentWithin(interval)
```

Comment:
if there is no startdate, or more importantly if there is no end date (if the employment is still active), the employment should still be returned

Reply:
👍